scipy/sparse/_sputils.py

```diff
@@ -130,7 +131,9 @@ def getdtype(dtype, a=None, default=None):
             raise ValueError(
                 "object dtype is not supported by sparse matrices"
             )
-
+    if newdtype not in supported_dtypes:
+        warn(f"scipy.sparse does not support dtype {newdtype}", stacklevel=3)
```

Comment:
Let's make this a ValueError. We can also remove the `np.object_` check above, as that will be caught by this new condition.

Reply:
Thanks for the tip! Fixed this and changed the error message slightly to include the supported dtypes.